File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/JavaCompileOptions.java
package com.tyron.builder.model;

import org.jetbrains.annotations.NotNull;

/**
 * Java compile options.
 */
public interface JavaCompileOptions {

    /**
     * @return the java compiler encoding setting.
     */
    @NotNull
    String getEncoding();

    /**
     * @return the level of compliance Java source code has.
     */
    @NotNull
    String getSourceCompatibility();

    /**
     * @return the Java version to be able to run classes on.
     */
    @NotNull
    String getTargetCompatibility();

    /** @return true if core library desugaring is enabled, false otherwise. */
    boolean isCoreLibraryDesugaringEnabled();
}

File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/v2/ide/JavaCompileOptions.kt
package com.tyron.builder.model.v2.ide

import com.tyron.builder.model.v2.AndroidModel

/**
 * Java compile options.
 *
 * @since 4.2
 */
interface JavaCompileOptions: AndroidModel {
    /**
     * @return the java compiler encoding setting.
     */
    val encoding: String

    /**
     * @return the level of compliance Java source code has.
     */
    val sourceCompatibility: String

    /**
     * @return the Java version to be able to run classes on.
     */
    val targetCompatibility: String

    /** @return true if core library desugaring is enabled, false otherwise.
     */
    val isCoreLibraryDesugaringEnabled: Boolean
}
